FestivalPlanner/loader/views.py

```diff
@@ -37,35 +41,186 @@ def get_festival_row(festival):
-            'theater_count_on_file': file_record_count(theaters_path()),
-            'screen_count': Screen.screens.count,
-            'screen_count_on_file': file_record_count(screens_path()),
-        }
         context = add_base_context(self.request, super().get_context_data(**kwargs))
-        context['title'] = 'Load Theater Data'
-        context['theater_items'] = theater_items
+        context['title'] = f'{self.action} Theater Data'
+        context['action'] = self.action
+        context['theater_items'] = get_theater_items()
         context['log'] = get_log(session)
         unset_log(session)
         return context
 
     def form_valid(self, form):
         session = self.request.session
-        form.load_theater_data(session)
+        if self.action == 'dump':
+            form.dump_theater_data(session)
+        elif self.action == 'load':
+            form.load_theater_data(session)
+        return HttpResponseRedirect('/theaters/theaters')
+
+
+class NewTheaterDataListView(ListView):
+    """
+    Class-based view to merge new theater data into the corresponding tables.
+    """
+    template_name = 'loader/new_screens.html'
+    http_method_names = ['get']
+    object_list = None
+    queryset = None
+    context_object_name = 'new_screen_items'
+    color_by_exists = {True: 'silver', False: 'MediumSpringGreen'}
+
+    def get_queryset(self):
+        session = self.request.session
+        if NewTheaterDataView.state_nr == 0:
+            initialize_log(session)
+        NewTheaterDataView.reset_new_theater_data()
+        new_cities = NewTheaterDataView.new_cities
+        new_theaters = NewTheaterDataView.new_theaters
+        new_screens = NewTheaterDataView.new_screens
+
+        # Load new theater data.
+        _ = CityLoader(session, file=new_cities_path()).load_new_objects(new_cities)
+        _ = TheaterLoader(session, file=new_theaters_path()).load_new_objects(new_theaters, foreign_objects=new_cities)
+        _ = ScreenLoader(session, file=new_screens_path()).load_new_objects(new_screens, foreign_objects=new_theaters)
+
+        # Create the new screen items list.
+        new_screen_items = [self.get_new_screen_item(screen) for screen in new_screens]
+
+        return sorted(
+            new_screen_items,
+            key=lambda new_screen_item: (new_screen_item['theater'], new_screen_item['screen_abbr'])
+        )
+
+    def get_context_data(self, *args, **kwargs):
+        context = add_base_context(self.request, super().get_context_data(**kwargs))
+        context['title'] = 'Merge New Screenings'
+        context['log'] = get_log(self.request.session)
+        context['objects_label'] = NewTheaterDataView.states[NewTheaterDataView.state_nr]
+        return context
+
+    def get_new_screen_item(self, screen):
+        new_screen_item = {
+            'city': screen.theater.city.name,
+            'city_color': self.color(City, City.cities, **{'country': 'nl', 'name': screen.theater.city.name}),
```

Comment:
See previous comment, could this be simplified from arguments `cls`, `manager` to a single argument `cls`?

Reply:
See my reply to that comment, #328 will do it.